obsreport/obsreport_receiver.go

```diff
@@ -68,19 +68,51 @@ var (
 		stats.UnitDimensionless)
 )
 
+// StartReceiveOptions has the options related to starting a receive operation.
+type StartReceiveOptions struct {
+	// LongLivedCtx is used in cases when individual receive messages happen
+	// under a single long lived context.
+	LongLivedCtx context.Context
```

Comment:
This should be just a boolean, the context passed in StartTraceDataReceiveOp will be long lived, we just need to know that is a long lived not a per op one.